src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2011,3 +1866,19 @@ Base.getindex(::AbstractDataFrame, ::Union{Symbol, Integer, AbstractString}) =
 
 Base.setindex!(::AbstractDataFrame, ::Any, ::Union{Symbol, Integer, AbstractString}) =
     throw(ArgumentError("syntax df[column] is not supported use df[!, column] instead"))
+
+# Iteration protocol
+
+function Base.iterate(df::AbstractDataFrame, i=1)
+    if i > nrow(df)
+        return nothing
+    else
+        return (df[i, :], i+1)
+    end
+end
+
+# make sure we disable potential specializations based on eltype for safety reasons
+Base.IteratorEltype(::Type{<:AbstractDataFrame}) = Base.EltypeUnknown()
```

Comment:
Shouldn't we require `DataFrameRow` as the eltype?

Reply:
We do not have to and `DataFrameRow` is abstract anyway so I did not see big benefit from overspecializing.